scripts/lib/CIME/case_run.py

```diff
@@ -92,7 +95,35 @@ def _run_model_impl(case, lid):
     logger.info("run command is %s " %cmd)
 
     rundir = case.get_value("RUNDIR")
-    run_cmd_no_fail(cmd, from_dir=rundir)
+    loop = True
+    num_spare_nodes = case.get_value("NUM_SPARE_NODES")
+    while loop:
+        loop = False
+        stat = run_cmd(cmd, from_dir=rundir)[0]
+        # Determine if failure was due to a failed node, if so, try to restart
+        if stat != 0:
+            node_fail_re = case.get_value("NODE_FAIL_RE")
+            if node_fail_re and num_spare_nodes > 0:
+                node_fail_regex = re.compile(node_fail_re)
+                model_logfile = os.path.join(rundir, model + ".log." + lid)
+                if os.path.exists(model_logfile):
+                    search = node_fail_regex.search(open(model_logfile, 'r').read())
+                    if search is not None:
+                        # We failed due to node failure!
+                        logger.warning("Detected model run failed due to node failure, restarting")
+
+                        case_st_archive(case, no_resubmit=True)
+
+                        case.set_value("CONTINUE_RUN", "TRUE")
```

Comment:
This won't work because there is no step to copy this xml value change into the model namelist where it is needed.   I think that you need to first check the current value of CONTINUE_RUN and if it's FALSE then you need to set it to try and run preview_namelists to update the model namelist files with the change.